packages/forms/frontend/sirius-components-widget-reference/src/modals/CreateModal.tsx

```diff
@@ -253,71 +234,55 @@ export const CreateModal = ({ editingContextId, widget, onClose }: CreateModalPr
     }
   }, [childCreationDescriptionsLoading, childCreationDescriptionsData, childCreationDescriptionsError]);
 
-  const [createChild, { loading: createChildLoading, data: createChildData, error: createChildError }] =
-    useMutation<GQLCreateChildMutationData>(createChildMutation);
+  const [createElement, { loading: createElementLoading, error: createElementError, data: createElementData }] =
+    useMutation<GQLCreateElementMutationData, GQLCreateElementMutationVariables>(createElementMutation);
 
   useEffect(() => {
-    if (!createChildLoading) {
-      if (createChildError) {
+    if (!createElementLoading) {
+      if (createElementError) {
         addErrorMessage('An unexpected error has occurred, please refresh the page');
       }
-      if (createChildData) {
-        const handleResponseEvent: HandleCreateChildResponseEvent = {
-          type: 'HANDLE_CHILD_CREATE_RESPONSE',
-          data: createChildData,
+      if (createElementData) {
+        const handleResponseEvent: HandleCreateElementResponseEvent = {
+          type: 'HANDLE_CREATE_ELEMENT_RESPONSE',
+          data: createElementData,
         };
         dispatch(handleResponseEvent);
 
-        const { createChild } = createChildData;
-        if (isErrorPayload(createChild)) {
-          const { message } = createChild;
+        const { createElement } = createElementData;
+        if (isErrorPayload(createElement)) {
+          const { message } = createElement;
           addErrorMessage(message);
         }
       }
     }
-  }, [createChildLoading, createChildData, createChildError]);
-
-  const [
-    createRootObject,
-    { loading: createRootObjectLoading, data: createRootObjectData, error: createRootObjectError },
-  ] = useMutation<GQLCreateRootObjectMutationData>(createRootObjectMutation);
-
-  useEffect(() => {
-    if (!createRootObjectLoading) {
-      if (createRootObjectError) {
-        addErrorMessage('An unexpected error has occurred, please refresh the page');
-      }
-      if (createRootObjectData) {
-        const handleResponseEvent: HandleCreateRootResponseEvent = {
-          type: 'HANDLE_ROOT_CREATE_RESPONSE',
-          data: createRootObjectData,
-        };
-        dispatch(handleResponseEvent);
-      }
-    }
-  }, [createRootObjectLoading, createRootObjectError, createRootObjectData]);
+  }, [createElementLoading, createElementData, createElementError]);
 
   const onCreateObject = () => {
+    let input = undefined;
```

Comment:
- The type of input is not defined as a result, you can create an input with a domainId a couple line below while the type `GQLCreateElementInput` does not allow domainId to be optional
- Use `null` instead of `undefined` to initialize variable to "the lack of a value"

Here use the type `let input: GQLCreateElementInput | null = null` and you will see that your type is wrong.

Reply:
Done